### DISCUSSION TITLE
Upgrade 25.0 fail in migration

### DISCUSSION THREAD

**Discussion by jeehou** [stashapp/stash#4669] [Q&A]
Migration failed
The following error was encountered while migrating the database:

Response not successful: Received status code 504

**To Reproduce**
1. upgrade the stash elf bin file to newone downloaded from github(https://github.com/stashapp/stash/releases/download/v0.25.0/stash-linux)
2. restart stash
3. open the web and automaticly saw migration page
4. click "Performer schema migration"

**Expected behavior**
migrate success 

**Screenshots**
![image](https://github.com/stashapp/stash/assets/1759238/960045b1-268b-458e-86fe-82392c557807)

**Stash Version: (from Settings -> About):**
upgrade from v24.3 to v25

**Desktop (please complete the following information):**
 - OS: ubuntu v20
 - Browser chrome
 - Version Version 121.0.6167.139 (Official Build) (arm64)

**Comment by WithoutPants** [comment]
Status code 504 is a Gateway Timeout error. As far as I know, stash does not return this status code in general, which suggests that you're using a reverse proxy or something else not covered in your description. Can you share more about your running environment?

**Comment by eddiebeazer** [comment]
By any chance are you using stash over a network share?  I had the same 504 and per @WithoutPants suggestion I accessed stash directly instead of via a reverse proxy.  The error I got was "Database is locked" instead of a 504.  Worth noting that stash worked perfectly fine on nfs but I had to switch because I was getting stale file handles.  Now I don't get those and instead have the db locking issue.  

Without going further off topic, more about your env would be good and in general, sqllite apps and network shares can really be hit or miss
